api/src/test/java/com/epam/pipeline/acl/docker/ToolApiServiceTest.java

```diff
@@ -0,0 +1,753 @@
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldUpdateToolForAdmin() {
+        doReturn(tool).when(mockToolManager).updateTool(tool);
+
+        assertThat(toolApiService.updateTool(tool)).isEqualTo(tool);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldUpdateToolWhenPermissionIsGranted() {
+        doReturn(tool).when(mockToolManager).loadTool(TEST_STRING, TEST_STRING);
+        doReturn(tool).when(mockToolManager).updateTool(tool);
+        initAclEntity(tool, AclPermission.WRITE);
+
+        doReturn(SecurityContextHolder.getContext().getAuthentication()).when(mockAuthManager).getAuthentication();
+        final Tool returnedTool = toolApiService.updateTool(tool);
+
+        assertThat(returnedTool.getMask()).isEqualTo(WRITE_PERMISSION);
+        assertThat(returnedTool).isEqualTo(tool);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldDenyUpdateToolWhenPermissionIsNotGranted() {
+        doReturn(tool).when(mockToolManager).loadTool(TEST_STRING, TEST_STRING);
+        initAclEntity(tool);
+
+        doReturn(SecurityContextHolder.getContext().getAuthentication()).when(mockAuthManager).getAuthentication();
+
+        assertThrows(AccessDeniedException.class, () -> toolApiService.updateTool(tool));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void updateWhiteListWithToolVersion() {
+        doReturn(toolVersionScanResult).when(mockToolManager)
+                .updateWhiteListWithToolVersionStatus(ID, TEST_STRING, true);
+
+        assertThat(toolApiService.updateWhiteListWithToolVersion(ID, TEST_STRING, true))
+                .isEqualTo(toolVersionScanResult);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyUpdateWhiteListWithToolVersion() {
+        doReturn(toolVersionScanResult).when(mockToolManager)
+                .updateWhiteListWithToolVersionStatus(ID, TEST_STRING, true);
+
+        assertThrows(AccessDeniedException.class,
+            () -> toolApiService.updateWhiteListWithToolVersion(ID, TEST_STRING, true));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadToolWithRegistryForAdmin() {
+        doReturn(tool).when(mockToolManager).loadTool(TEST_STRING, TEST_STRING);
+
+        assertThat(toolApiService.loadTool(TEST_STRING, TEST_STRING)).isEqualTo(tool);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadToolWithoutRegistryForAdmin() {
+        doReturn(tool).when(mockToolManager).loadByNameOrId(TEST_STRING);
+
+        assertThat(toolApiService.loadTool("", TEST_STRING)).isEqualTo(tool);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldLoadToolWithRegistryWhenPermissionIsGranted() {
+        initAclEntity(tool, AclPermission.READ);
+        doReturn(tool).when(mockToolManager).loadTool(TEST_STRING, TEST_STRING);
+
```

Comment:
I would rather not divide this logic: `doReturn(SecurityContextHolder.getContext().getAuthentication()).when(mockAuthManager).getAuthentication();` is another mock in this test.